### PR TITLE
added some components needed for sender concepts

### PR DESCRIPTION
**Pull request by dietmarkuehl**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (JeffGarland) [inline Makefile:2]: Is the plan to do cmake stuff later?

File: include/Beman/Execution26/detail/fwd_env.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    private:
+        Env env;
+    
+    public:
+        fwd_env(Env&& env): env(::std::forward<Env>(env)) {}
+
+        template <typename Query, typename... Args>
+            requires (not ::Beman::Execution26::forwarding_query(::std::remove_cvref_t<Query>()))
+        constexpr auto query(Query&& q, Args&&... args) const
+            = BEMAN_EXECUTION26_DELETE("the used query is not forwardable");
```
Comment (JeffGarland) [inline include/Beman/Execution26/detail/fwd_env.hpp:28]: delete with a message -- implemented with compatibility?

File: include/Beman/Execution26/detail/sender_decompose.hpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    template <typename Sender>
+    auto get_sender_meta(Sender&& sender)
+    {
+        #if 0
+        //-dk:TODO should use a dynamic/language approach:
+        auto&& [tag, data, ... children] = sender;
+        return sender_meta<decltype(tag), decltype(data), ::std::tuple<decltype(children)...>>;
+        #endif
+        using sender_type = ::std::remove_cvref_t<Sender>;
+        static constexpr ::Beman::Execution26::Detail::sender_any_t any{};
```
Comment (JeffGarland) [inline include/Beman/Execution26/detail/sender_decompose.hpp:45]: obviously we're in a different namespace here, but would we want to avoid the name any -- just to not confuse with std::any?  Or maybe this is actually just a temporary shim?

File: include/Beman/Execution26/detail/sender_in.hpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <Beman/Execution26/detail/empty_env.hpp>
+#include <Beman/Execution26/detail/get_completion_signatures.hpp>
+#include <Beman/Execution26/detail/queryable.hpp>
+#include <Beman/Execution26/detail/sender.hpp>
+#include <Beman/Execution26/detail/valid_completion_signatures.hpp>
+
+// ----------------------------------------------------------------------------
+
+namespace Beman::Execution26
```
Comment (JeffGarland) [inline include/Beman/Execution26/detail/sender_in.hpp:15]: ::Detail?  the file is under the directory?

File: Makefile (hunk 1 above)
Comment (JeffGarland) [reply Makefile:2]: hmm ok there's a cmake file later....

**Comment by JeffGarland** [comment]
I took a pass thru -- didn't see too much except maybe on missing namespace. It sure takes a lot of infrastructure to get anywhere with this proposal lol.

File: Makefile (hunk 1 above)
Comment (dietmarkuehl) [reply Makefile:2]: I'm using `make` to drive my builds even if the main work is done by `cmake`: the default target just runs `cmake` to build and test things. There are a few other target like a naïve dependency checker and something to build one header suitable for Compiler Explorer: I'd forget how to run these if I don't encode the "procedure" somewhere.

tl;dr: it uses `cmake` already! `make` is just a "convenience" frontend you don't have to use.

File: include/Beman/Execution26/detail/fwd_env.hpp (hunk 2 above)
Comment (dietmarkuehl) [reply include/Beman/Execution26/detail/fwd_env.hpp:28]: Well, [`delete` with message](http://wg21.link/p2573)  isn't implemented in any of the compilers I'm actually using (it seems it is coming with [gcc](http://gcc.gnu.org/)-15 and [clang](http://llvm.org)-19). I should probably build a current version and see if it actually works.

File: include/Beman/Execution26/detail/sender_in.hpp (hunk 4 above)
Comment (dietmarkuehl) [reply include/Beman/Execution26/detail/sender_in.hpp:15]: This comment is interesting: `sender_in` is supposed to be in `std::execution`. However, the file `sender_in.hpp` isn't anything the standard defines: the header is `<execution>` which is modelled by `include/Beman/Execution26/execution.hpp` which would eventually include `sender_in.hpp`.

I can't cope with creating just one header [manually], i.e., I'm organising my implementation mostly into one component per header. The header name is an implementation-specific detail, the name within is (in this case) a standard name. As a result, the namespace doesn't quite line up with the path.

Maybe the proper way is to actually define `sender_in` in the `Detail` namespace and get it into the standard namespace using a suitable alias (and, obviously, do so far other names, too). That sounds like a reasonable thing to do and I'll make a pass through the code to change it accordingly.

**Comment by dietmarkuehl** [comment]
> I took a pass thru -- didn't see too much except maybe on missing namespace. It sure takes a lot of infrastructure to get anywhere with this proposal lol.

Thank you for having a look!

Although already quite big, this PR isn't even done, yet! I'm primarily using it to run the github CI to make sure the code compiles with [clang](http://llvm.org/), [gcc](http://gcc.gnu.org), and [MSVC++](https://visualstudio.microsoft.com/vs/features/cplusplus/).

With respect to "a lot of infrastructure": despite all that code, I don't even have _any_ sender working at all! I know because `connect` doesn't have all its dependencies resolved and, thus, isn't implemented, yet (on the other hand, I have some things implemented which aren't required for `connect`).
